modules/apps/translation/translation-web/src/main/java/com/liferay/translation/web/internal/display/context/ImportTranslationResultsDisplayContext.java

```diff
@@ -66,6 +73,35 @@ public int getFailureMessagesCount() {
 		return _failureMessages.size();
 	}
 
+	public String getFailureMessagesCSVDataURL(Locale locale) {
+		String csv = "";
+
+		try {
+			StringWriter stringWriter = new StringWriter();
+
+			CSVPrinter csvPrinter = new CSVPrinter(
+				stringWriter,
+				CSVFormat.DEFAULT.withHeader(
+					LanguageUtil.get(locale, "file-name"),
+					LanguageUtil.get(locale, "error-message")));
+
+			for (Map.Entry<String, String> entry :
+					_failureMessages.entrySet()) {
+
+				csvPrinter.printRecord(entry.getKey(), entry.getValue());
+			}
+
+			csv = stringWriter.toString();
+		}
+		catch (Exception exception) {
+			if (_log.isDebugEnabled()) {
+				_log.debug(exception, exception);
+			}
+		}
+
+		return "data:text/csv;charset=utf-8," + JS.encodeURIComponent(csv);
+	}
+
```

Comment:
If it fails, let it fail naturally, don't swallow errors we cannot handle;  returning an empty value on failure looks wrong to me. Expose the exception type (not Exception; as a last resort use PortalException, wrapping the original failure).

Also, avoid having unnecessary variables. If you let it fail, you can directly return the result of `stringWriter.toString()`.

Reply:
That makes sense, @AliciaGarciaGarcia I'll need your help with this :D